Video file name with double dot right can't be delivered

<!-- Delete/replace placeholder texts -->

**Describe the bug**

When uploading a video with a double dot (e.g. `Big_Buck_Bunny_first_23_seconds_1080p..mp4`) can't be delivered from engage with secure static files turned on in spite being logged in as an administrator. The error message is:

<h2>HTTP ERROR 403 Forbidden</h2>

URI: | /static/mh_default_org/engage-player/f5bfba16-7b61-4573-99a0-142f940108a2/b208388d-849c-49f2-8f6a-378ac349bece/Big_Buck_Bunny_first_23_seconds_1080p..mp4
-- | --
STATUS: | 403
MESSAGE: | Forbidden
SERVLET: | org.ops4j.pax.web.service.spi.model.ServletModel-11

On distributed setups, the public-engage step fails with:

```
org.opencastproject.serviceregistry.api.ServiceRegistryException: Error handling operation 'Distribute'
	at org.opencastproject.distribution.download.DownloadDistributionServiceImpl.process(DownloadDistributionServiceImpl.java:714)
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:313)
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:272)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

```
Caused by: org.opencastproject.distribution.api.DistributionException: Unable to load distributed file https://presentation.opencast.example.com/static/mh_default_org/engage-player/929c90b2-4ccc-4c17-a3c4-7048e9eb7840/328e347e-6d95-4f5a-87dc-acedc9d997f6/2023-01-25_22-01_-_Studienleistung_Projektarbeit_-_Stella_Gebhardt__A..m4a
    at org.opencastproject.distribution.download.DownloadDistributionServiceImpl$3.xrun(DownloadDistributionServiceImpl.java:884)
    at org.opencastproject.distribution.download.DownloadDistributionServiceImpl$3.xrun(DownloadDistributionServiceImpl.java:879)
    at org.opencastproject.util.data.Effect$X.run(Effect.java:47)
    at org.opencastproject.util.data.Effect.apply(Effect.java:35)
    at org.opencastproject.util.data.Effect$X.apply(Effect.java:43)
    at org.opencastproject.util.data.Either$2.fold(Either.java:260)
    at org.opencastproject.distribution.download.DownloadDistributionServiceImpl.lambda$checkAvailability$7(DownloadDistributionServiceImpl.java:879)
    at org.opencastproject.security.util.SecurityUtil.runAs(SecurityUtil.java:79)
    at org.opencastproject.distribution.download.DownloadDistributionServiceImpl.checkAvailability(DownloadDistributionServiceImpl.java:877)
    at org.opencastproject.distribution.download.DownloadDistributionServiceImpl.distributeElement(DownloadDistributionServiceImpl.java:370)
    at org.opencastproject.distribution.download.DownloadDistributionServiceImpl.distributeElements(DownloadDistributionServiceImpl.java:287)
    at org.opencastproject.distribution.download.DownloadDistributionServiceImpl.process(DownloadDistributionServiceImpl.java:698)
    ... 6 more
```

**To Reproduce**

Steps to reproduce the behavior:
1. Go to https://stable.opencast.org
2. Click on +Add New Event
3. Fill-in all the required fields and proceed to the file selection
4. Select any sane video file you find, but before uploading, rename it to something with two consecutive dots before their file name extension (e.g. `test..mp4`)
5. Proceed and finish the upload
6. Wait until the video is fully processed
7. Open the engage link ![image](https://user-images.githubusercontent.com/2311611/214852312-f6f33e05-65f1-410a-8937-8bd738fb4bb8.png)
8. Press Play and see the never ending circle (IHMO Paella should also handle this more gracefully) ![image](https://user-images.githubusercontent.com/2311611/214852660-9f2c7b68-16da-4783-8e57-66a71817687a.png)
9. Inspect the network traffic and reload. Open the the video failed loading in a new tab ![image](https://user-images.githubusercontent.com/2311611/214852080-907ac99f-5307-4920-b298-6dbf42b356af.png)


**Expected behavior**

I can watch the video after publication the same way I can with any videos that do not contain a double-dot.

**Server environment:**
- Distribution: all (distributed setups are additionally hit by the failed publish step)
- Installation method: all
- Version r/12.x r/13.x (14 not tested)

--- a/etc/org.opencastproject.workingfilerepository.impl.WorkingFileRepository.cfg
+++ b/etc/org.opencastproject.workingfilerepository.impl.WorkingFileRepository.cfg
@@ -4,18 +4,19 @@
 # and file systems and to prevent local security risks like path traversal. For example, the default conversion would
 # look like this:
 #
-#   original:  -problematic-file/../name&&&%.ext
-#   converted: _problematic-file_.._name____.ext
+#   original:  -problem&&&%atic-file.highres/../name..ext
+#   converted: _problem____atic-file.highres___name_.ext
 #
 # This option allows you to control which characters are not allowed. All parts of the file name matched by the
 # configured pattern will be replaced by `_`. The replacement is applied separately to the base file name and the
 # extension. In any case, both base name and extension are limited to 255 characters each.
 #
 # WARNING: This setting is dangerous. It lets you configure Opencast in a way that poses a security risk to your system
-#          by e.g. allowing for path traversal attacks.
+#          by e.g. allowing for path traversal attacks. Note that allowing dot characters can lead to paths, which are
+#          not allowed to be served (i.e. containing "..").
 # BETA:    For a long time, this was a very restricted, fixed value. Consider this to be an unstable configuration which
 #          might lead to problems since some component somewhere assumes the old restricted behavior. Test your system
 #          very carefully after changing this. Feedback about problems and successful changes are welcome.
 #
-# Default: (^\\W|[^\\w-.])
-#filename.forbidden.pattern = (^\\W|[^\\w-_.])
+# Default: (^\\W|[^\\w-.]|\\.\\.|\\.$)
+#filename.forbidden.pattern = (^\\W|[^\\w-.]|\\.\\.|\\.$)

--- a/modules/working-file-repository-service-impl/src/main/java/org/opencastproject/workingfilerepository/impl/WorkingFileRepositoryImpl.java
+++ b/modules/working-file-repository-service-impl/src/main/java/org/opencastproject/workingfilerepository/impl/WorkingFileRepositoryImpl.java
@@ -118,7 +118,7 @@ public class WorkingFileRepositoryImpl implements WorkingFileRepository, PathMap
   protected String servicePath = null;
 
   /** The default pattern for characters forbidden in filenames */
-  private static final String FILENAME_REGEX_DEFAULT = "(^\\W|[^\\w-.])";
+  private static final String FILENAME_REGEX_DEFAULT = "(^\\W|[^\\w-.]|\\.\\.|\\.$)";
 
   /** Key for configuring the filename pattern specifying forbidden characters */
   private static final String FILENAME_REGEX_KEY = "filename.forbidden.pattern";

